uber/config.py

```diff
@@ -165,8 +174,33 @@ def SUPPORTERS_ENABLED(self):
 
     @property
     def AT_THE_DOOR_BADGE_OPTS(self):
-        opts = [(self.ATTENDEE_BADGE, 'Full Weekend Pass (${})'.format(self.BADGE_PRICE))]
-        if self.ONE_DAYS_ENABLED:
+        """
+        This populates the dropdown form on the /registration/register page with a list of badges available at-door.
+        It always offers a "Full Weekend Pass" if attendee badges are available.
+        If one-days are enabled, it will show either a generic "Single Day Pass" or a list of specific day badges,
+        depending on if PRESELL_ONE_DAYS is true. Each day's badge will need to be configured under [badge_ranges] and
+        [[badge]] to show up in this dropdown.
+        """
+        opts = []
+        if self.ATTENDEE_BADGE_AVAILABLE:
+            opts.append((self.ATTENDEE_BADGE, 'Full Weekend Pass (${})'.format(self.BADGE_PRICE)))
+        if self.ONE_DAYS_ENABLED and self.PRESELL_ONE_DAYS:
+            iterdate = max(sa.localized_now(), self.EPOCH)
+            while iterdate.date() <= self.ESCHATON.date():
+                price = self.BADGE_PRICES['single_day'].get(iterdate.strftime('%A'))
+                if price:
+                    try:
+                        badge = getattr(c, iterdate.strftime('%A').upper() + "_BADGE")
+                    except AttributeError:
+                        log.error('Could not add badge to AT_THE_DOOR_BADGE_OPTS. '
+                                  'Badge type not configured: {}', iterdate.strftime('%A').upper() + '_BADGE')
+                        iterdate += timedelta(days=1)
+                        continue
+
+                    if getattr(c, iterdate.strftime('%A').upper() + "_BADGE_AVAILABLE", None):
+                        opts.append((badge, iterdate.strftime('%A') + ' (${})'.format(price)))
+                iterdate += timedelta(days=1)
```

Comment:
Looks great!  Minor style suggestion: do ``try/except/else`` here to avoid needing to say ``continue`` and having to increment ``iterdate`` in 2 places.  In other words:

```python
    if price:
        try:
            badge = getattr(c, iterdate.strftime('%A').upper() + "_BADGE")
        except AttributeError:
            log.error('Could not add badge to AT_THE_DOOR_BADGE_OPTS. '
                      'Badge type not configured: {}', iterdate.strftime('%A').upper() + '_BADGE')
        else:
            if getattr(c, iterdate.strftime('%A').upper() + "_BADGE_AVAILABLE", None):
                opts.append((badge, iterdate.strftime('%A') + ' (${})'.format(price)))
    iterdate += timedelta(days=1)
```